src/Bridge/Doctrine/Orm/Extension/PaginationExtension.php

```diff
@@ -33,28 +32,18 @@
     private $managerRegistry;
     private $requestStack;
     private $resourceMetadataFactory;
-    private $enabled;
-    private $clientEnabled;
-    private $clientItemsPerPage;
-    private $itemsPerPage;
-    private $pageParameterName;
-    private $enabledParameterName;
-    private $itemsPerPageParameterName;
-    private $maximumItemPerPage;
-
-    public function __construct(ManagerRegistry $managerRegistry, RequestStack $requestStack, ResourceMetadataFactoryInterface $resourceMetadataFactory, bool $enabled = true, bool $clientEnabled = false, bool $clientItemsPerPage = false, int $itemsPerPage = 30, string $pageParameterName = 'page', string $enabledParameterName = 'pagination', string $itemsPerPageParameterName = 'itemsPerPage', int $maximumItemPerPage = null)
+    private $paginationHelperFactory;
+
+    public function __construct(ManagerRegistry $managerRegistry, RequestStack $requestStack, ResourceMetadataFactoryInterface $resourceMetadataFactory, bool $enabled = true, bool $clientEnabled = false, bool $clientItemsPerPage = false, int $itemsPerPage = 30, string $pageParameterName = 'page', string $enabledParameterName = 'pagination', string $itemsPerPageParameterName = 'itemsPerPage', int $maximumItemPerPage = null, PaginationHelperFactory $paginationHelperFactory = null)
     {
         $this->managerRegistry = $managerRegistry;
         $this->requestStack = $requestStack;
         $this->resourceMetadataFactory = $resourceMetadataFactory;
-        $this->enabled = $enabled;
-        $this->clientEnabled = $clientEnabled;
-        $this->clientItemsPerPage = $clientItemsPerPage;
-        $this->itemsPerPage = $itemsPerPage;
-        $this->pageParameterName = $pageParameterName;
-        $this->enabledParameterName = $enabledParameterName;
-        $this->itemsPerPageParameterName = $itemsPerPageParameterName;
-        $this->maximumItemPerPage = $maximumItemPerPage;
+
+        if (!$paginationHelperFactory) {
```

Comment:
I would write: `$this->paginationHelperFactory = $paginationHelperFactory ?? new PaginationHelperFactory($requestStack, $resourceMetadataFactory, $enabled, $clientEnabled, $clientItemsPerPage, $itemsPerPage, $pageParameterName, $enabledParameterName, $itemsPerPageParameterName, $maximumItemPerPage);`